applications/fishing-map/src/features/datasets/datasets.slice.ts

```diff
@@ -85,10 +86,12 @@ export const fetchDatasetsByIdsThunk = createAsyncThunk(
         `/v1/datasets?${stringify(workspacesParams, { arrayFormat: 'comma' })}`,
         { signal }
       )
-      const relatedDatasetsIds = initialDatasets.flatMap(
-        (dataset) => dataset.relatedDatasets?.flatMap(({ id }) => id || []) || []
+      const relatedDatasetsIds = uniq(
+        initialDatasets.flatMap(
+          (dataset) => dataset.relatedDatasets?.flatMap(({ id }) => id || []) || []
+        )
       )
-      const uniqRelatedDatasetsIds = without(relatedDatasetsIds, ...ids).join(',')
+      const uniqRelatedDatasetsIds = without(relatedDatasetsIds, ...existingIds).join(',')
```

Comment:
Also, this fixes a bug causing an [API 431 error](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/431) when too many datasets